packages/video_player/video_player_android/example/integration_test/video_player_test.dart

```diff
@@ -98,7 +98,9 @@ void main() {
     final Duration pausedDuration = await player.getPosition(textureId);
     await tester.pumpAndSettle(_playDuration);
 
-    expect(await player.getPosition(textureId), pausedDuration);
+    // TODO(camsim99): Implement reliable way to tell when pausing completes
+    // to check for exact pausedDuration.
+    expect((await player.getPosition(textureId)).inMilliseconds, inInclusiveRange(pausedDuration.inMilliseconds, pausedDuration.inMilliseconds + 500));
```

Comment:
I am legit confused why this is necessary, the original code (before the refactor) read:

```dart
      await controller.initialize();

      // Play for a second, then pause, and then wait a second.
      await controller.play();
      await tester.pumpAndSettle(_playDuration);
      await controller.pause();
      await tester.pumpAndSettle(_playDuration);
      final Duration pausedPosition = (await controller.position)!;
      await tester.pumpAndSettle(_playDuration);

      // Verify that we stopped playing after the pause.
      expect(await controller.position, pausedPosition);
```

Maybe that intermediate `pumpAndSettle` is necessary?

Reply:
Ah I didn't realize we removed one! Added it back to see.

Reply:
Thank you! That was my bad as well, I wish there was an easy to way run a test 100 times.